platforms/jvm/jacoco/src/integTest/groovy/org/gradle/testing/jacoco/plugins/JacocoConfigurationCacheIntegrationTest.groovy

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.testing.jacoco.plugins
+
+import org.gradle.integtests.fixtures.AbstractIntegrationSpec
+import org.gradle.test.fixtures.dsl.GradleDsl
+import org.gradle.test.precondition.Requires
+import org.gradle.test.preconditions.IntegTestPreconditions
+import spock.lang.Issue
+
+@Requires(value = IntegTestPreconditions.NotConfigCached, reason = 'Test explicitly enables the configuration cache')
+class JacocoConfigurationCacheIntegrationTest extends AbstractIntegrationSpec {
+
+    def setup() {
+        // run all tests with configuration caching
+        file('gradle.properties') << 'org.gradle.configuration-cache=true'
+    }
+
+    @Issue('https://github.com/gradle/gradle/issues/26922')
+    def 'can aggregate with `kotlin-dsl` subproject'() {
+        given:
+        file('settings.gradle.kts') << """
+            include(":plugin")
+            dependencyResolutionManagement {
+                ${mavenCentralRepository(GradleDsl.KOTLIN)}
+            }
+        """
+        file('build.gradle.kts') << '''
+            plugins { id("jacoco-report-aggregation") }
+
+            reporting {
+                reports {
+                    val testCodeCoverageReport by creating(JacocoCoverageReport::class) {
+                        testType = TestSuiteType.UNIT_TEST
+                    }
+                }
+            }
+
+            dependencies {
+                jacocoAggregation(project(":plugin"))
+            }
+        '''
+        createDir('plugin') {
+            file('build.gradle.kts') << '''
+                plugins { `kotlin-dsl` }
```

Comment:
Do we really need to apply `kotlin-dsl` to replicate this issue? 

The linked issue states:

"When changing plugin kotlin-dsl to kotlin-dsl-base in a project that is added to jacocoAggregation, the build also completes successfully with the configuration cache."

What happens between kotlin-dsl and kotlin-dsl-base that causes this problem? Can we test that directly?

Reply:
Yes, `kotlin-dsl` is needed.

I haven't investigated what's the difference between the two plugins that could explain the problem but I think that can be done after the fix has been merged.

Reply:
The issue is actually triggered by `java-gradle-plugin`. It seems the publications it creates cause the component filters to be necessary (thus serialized to the configuration cache).

Reply:
Can we go deeper and manually reproduce what `java-gradle-plugin` does?

I'm worried that if the java-gradle-plugin implementation changes that this test will no longer actually test what we want

Reply:
That makes sense. I don't have the bandwidth right now though so I'm tempted to merge this as is (after adding an issue to the cc backlog to review the test coverage).

Reply:
Thanks

Reply:
FTR: [the issue](https://github.com/orgs/gradle/projects/63/views/7?filterQuery=subproject%3A%22Adoption%22+jacoco&pane=issue&itemId=46923389)